atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2484,45 +2465,11 @@ private boolean rollbackFailedTransactions(
         });
 
         if (allRolledBack) {
-            try {
-                runTaskOnDeleteExecutor(kvs -> deleteCells(kvs, tableRef, keysToDelete));
-            } catch (RejectedExecutionException rejected) {
-                deleteExecutorRateLimitedLogger.log(logger -> logger.info(
-                        "Could not delete keys {} for table {}, because the delete executor's queue was full."
-                                + " Sweep should eventually clean these values.",
-                        SafeArg.of("numKeysToDelete", keysToDelete.size()),
-                        LoggingArgs.tableRef(tableRef),
-                        rejected));
-            }
+            deleteExecutor.scheduleForDeletion(tableRef, keysToDelete);
```

Comment:
It is worth noting that we assume this collection of keys passed here IS NOT GOING to change. The way this code has been written historically this is the case, and I don't think I'm breaking that.